habitat-lab/habitat/config/CONFIG_KEYS.md

```diff
@@ -0,0 +1,88 @@
+# Configuration Keys
+
+## Dataset
+Configuration for the dataset of the task.
+A dataset consists of episodes
+(a start configuration for a task within a scene) and a scene dataset
+(with all the assets needed to instantiate the task)
+
+| Key | Description |
+| --- | --- |
+| habitat.dataset.type |  The key for the dataset class that will be used. Examples of such keys are `PointNav-v1`, `ObjectNav-v1`, `InstanceImageNav-v1` or `RearrangeDataset-v0`. Different datasets have different properties so you should use the dataset that fits your task. |
+| habitat.dataset.scene_dir | The path to the directory containing the scenes that will be used. You should put all your scenes in the same folder (example `data/scene_datasets`) to avoid having to change it. |
+|habtat.dataset.data_path | The path to the episode dataset. Episodes need to be compatible with the `type` argument (so they will load properly) and only use scenes that are present in the `scenes_dir`.|
+|habitat.dataset.split | `data_path` can have a `split` in the path. For example: "data/datasets/pointnav/habitat-test-scenes/v1/{split}/{split}.json.gz" the value in "{split}" will be replaced by the value of the `split` argument. This allows to easily swap between `train` and `eval` episodes by only changing the split argument.|
+
+## Task
+The definition of the task in Habitat.
+
+| Key | Description |
+| --- | --- |
+|habitat.task.type | The registered task that will be used. For example : `InstanceImageNav-v1` or `ObjectNav-v1`
```

Comment:
What is the relation between `dataset.type` and `task.type`?

Reply:
Ambiguous. In navigation the dataset and the task usually have the same name but some tasks use the same dataset. In rearrangement there is only one RearrangeDataset-v0 type.